lib/oc-tests/test_dirDepthDownload.py

```diff
@@ -0,0 +1,173 @@
+|  4        | final step                                          |
++-----------+-----------------------------------------------------+
+
+Data Providers:
+
+  test_sharePermissions:      Permissions to be applied to the share
+  test_numFilesToCreate:      Number of files to create
+  test_filesizeKB:            Size of file to create in KB
+  dir_depth:                  How deep the directory structure should go
+  style:                      Defines if the directory layout is flat or hierarchial
+  num_file_rows:              How many rows in the file
+
+
+"""
+
+from smashbox.utilities import *
+import glob
+import re
+
+filesizeKB = int(config.get('test_filesizeKB',10))
+numFilesToCreate = config.get('test_numFilesToCreate', 10)
+dir_depth = config.get ('dir_depth',5)
+num_users = config.get ('oc_number_test_users', 3)
+num_file_rows = config.get ('file_row_count', 100)
+style = config.get ('dir_depth_style', 'complex')
+
+testsets = [
+    {
+        'num_file_rows':1000,
+        'num_users':10,
+        'dir_depth':5,
+        'test_numFilesToCreate':50,
+        'test_filesizeKB':20,
+        'style': 'flat',
+    },
+    {
+        'num_file_rows':1000,
+        'num_users':10,
+        'dir_depth':5,
+        'test_numFilesToCreate':50,
+        'test_filesizeKB':200,
+        'style': 'complex',
+    },
+    {
+        'num_file_rows':1000,
+        'num_users':10,
+        'dir_depth':10,
+        'test_numFilesToCreate':50,
+        'test_filesizeKB':2000,
+        'sytle': 'flat'
+    },
+    {
+        'num_file_rows':1000,
+        'num_users':10,
+        'dir_depth':10,
+        'test_numFilesToCreate':50,
+        'test_filesizeKB':2000,
+        'sytle': 'complex'
+    },
+]
+
+@add_worker
+def setup(step):
+
+    step (1, 'Create test users')
+    reset_owncloud_account(num_test_users=num_users)
+    check_users(num_users)
+
+    reset_rundir()
+    reset_server_log_file()
+
+    step (5, 'Validate server log file is clean')
+
+    d = make_workdir()
+    scrape_log_file(d)
+
+
+def uploader (step):
+
+    step (2,'Create workdir')
+    d = make_workdir()
+    uploader_num = get_user_number_from_work_directory(d)
+
+    step (3,'Create directories and files then sync and validate')
+
+    filename = "%s%s" % (d,'/TEST_FILE_NEW_USER_SHARE.dat')
+    createfile(filename,'0',count=1000,bs=filesizeKB)
+    upload_dir = "%s%d" % ('uploader', uploader_num)
+
+    procName = reflection.getProcessName()
+
+    if style is 'flat':
+        for i in range(dir_depth):
+            upload_dir = "%s_%d"%('upload_dir', i) 
+            webdav_mkcol (upload_dir, user_num = uploader_num)
+            for j in range(0, numFilesToCreate):
+                upload_name = "%s_%d.dat" % ('TEST_FILE_NEW_USER_SHARE', j)
+                webdav_upload(filename, upload_dir, upload_name, uploader_num)
+    else:
+        upload_dir = procName
+        webdav_mkcol (upload_dir, user_num = uploader_num)
```

Comment:
It seems to me there is code duplication here... Is there any other way to do this? (this "if - else" block)